highway_env/envs/parking_env.py

```diff
@@ -131,60 +141,152 @@ def _create_road(self, spots: int = 14) -> None:
 
         :param spots: number of spots in the parking
         """
+        spots = self.config["spots"]
         net = RoadNetwork()
         width = 4.0
         lt = (LineType.CONTINUOUS, LineType.CONTINUOUS)
         x_offset = 0
-        y_offset = 10
+        y_offset = self.config["y_offset"]
         length = 8
-        for k in range(spots):
-            x = (k + 1 - spots // 2) * (width + x_offset) - width / 2
-            net.add_lane("a", "b", StraightLane([x, y_offset], [x, y_offset+length], width=width, line_types=lt))
-            net.add_lane("b", "c", StraightLane([x, -y_offset], [x, -y_offset-length], width=width, line_types=lt))
+
+        row_ys = []
+        for i in range(self.config["n_rows"]):
+            if i == 0:
+                row_ys.append([y_offset, y_offset+length])
+            elif i == 1:
+                row_ys.append([-y_offset, -y_offset-length])
+            else:
+                if i%2 == 0:
+                    row_ys.append([row_ys[i-2][1]+2*y_offset, row_ys[i-2][1]+2*y_offset+length])
+                else:
+                    row_ys.append([row_ys[i-2][1]-2*y_offset, row_ys[i-2][1]-2*y_offset-length])
+
+        id = 0
+        for row in range(0, self.config["n_rows"]):
+            for k in range(spots):
+                x = (k + 1 - spots // 2) * (width + x_offset) - width / 2
+                net.add_lane(
+                    chr(ord('a')+row), chr(ord('a')+row+1), 
+                    StraightLane([x, row_ys[row][0]], [x, row_ys[row][1]], 
+                                 width=width, line_types=lt, identifier=id, 
+                                 display_font_size=self.config['font_size']
+                    )
+                )
+                id += 1
 
         self.road = Road(network=net,
                          np_random=self.np_random,
                          record_history=self.config["show_trajectories"])
+        
+        # Store the allowed x,y coordinate ranges to spawn the agent
+        allowed_y_space = [[-y_offset+8, y_offset-8]]
+        if len(row_ys) > 2:
+            for i in range(2, len(row_ys)):
+                if i%2 == 0:
+                    allowed_y_space.append([row_ys[i-2][1]+8, row_ys[i][0]-8])
+                else:
+                    allowed_y_space.append([row_ys[i][0]+8, row_ys[i-2][1]-8])
+
+        self.allowed_vehicle_space = {
+            'x' : (-30, 30),
```

Comment:
Not sure why this is hardcoded, it doesn't seem to adapt to other configs. Also, this is not used here, please move to create_vehicles.